scripts/pre_commit_linter.py

```diff
@@ -340,21 +362,49 @@ class FileCache(object):
 
     @classmethod
     def read(cls, filepath, mode='r'):
-        """Returns the data read from the file.
+        """Returns the data read from the file in unicode form.
 
         Args:
             filepath: str. The file path from which data is to be read.
             mode: str. The mode in which the file is to be opened.
 
         Returns:
-            str. The data read from the file.
+            unicode. The data read from the file.
         """
         return cls._get_data(filepath, mode)[0]
 
+    @classmethod
+    def read_as_str(cls, filepath, mode='r'):
+        """Returns the data read from the file in str form.
+
+        Args:
+            filepath: str. The file path from which data is to be read.
+            mode: str. The mode in which the file is to be opened.
+
+        Returns:
+            str. The data read from the file.
+        """
+        return utils.convert_to_str(cls._get_data(filepath, mode)[0])
+
     @classmethod
     def readlines(cls, filepath, mode='r'):
         """Returns the tuple containing data line by line as read from the
-        file.
+        file in unicode form.
+
+        Args:
+            filepath: str. The file path from which data is to be read.
+            mode: str. The mode in which the file is to be opened.
+
+        Returns:
+            tuple(unicode). The tuple containing data line by line as read from
+                the file.
+        """
+        return cls._get_data(filepath, mode)[1]
+
+    @classmethod
+    def readlines_as_str(cls, filepath, mode='r'):
```

Comment:
Ditto. Transform to unicode at data boundaries and work with unicode internally throughout; there should be no conversions of str <--> unicode in the interior of the file. Otherwise it gets confusing because sometimes you are dealing with str and sometimes you are dealing with unicode.

Reply:
Addressed in the above comment

Reply:
Done